lib/model.js

```diff
@@ -1411,6 +1411,42 @@ Model.syncIndexes = function syncIndexes(options, callback) {
   }, this.events);
 };
 
+
+Model.diffIndexes = function diffIndexes(options, callback) {
+  const toDrop = [];
+  const toCreate = [];
+  callback = this.$handleCallbackError(callback);
+  return this.db.base._promiseOrCallback(callback, cb => {
+    cb = this.$wrapCallback(cb);
+    this.listIndexes((err, indexes) => {
+      const schemaIndexes = this.schema.indexes();
+      for (const index of indexes) {
+        let found = false;
+        // Never try to drop `_id` index, MongoDB server doesn't allow it
+        if (isDefaultIdIndex(index)) {
+          continue;
+        }
+
+        for (const schemaIndex of schemaIndexes) {
+          const key = schemaIndex[0];
+          const options = _decorateDiscriminatorIndexOptions(this,
+            utils.clone(schemaIndex[1]));
+          if (isIndexEqual(key, options, index)) {
+            found = true;
+          } else {
```

Comment:
You may end up adding the same index to `toCreate` multiple times. If you have 3 indexes in MongoDB, you'll execute the `schemaIndexes()` loop 3 times and end up adding the same schema index twice each time.

You would likely need to write a separate `for` loop to correctly compute `toCreate`